client/my-sites/plans/hooks/use-plan-prices.ts

```diff
@@ -0,0 +1,36 @@
+import { getPlan, PlanSlug } from '@automattic/calypso-products';
+import { useSelector } from 'react-redux';
+import { getPlanRawPrice, getDiscountedRawPrice } from 'calypso/state/plans/selectors';
+import { getPlanDiscountedRawPrice } from 'calypso/state/sites/plans/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+export interface PlanPrices {
+	rawPrice: number;
+	discountedRawPrice: number; // discounted on yearly-monthly conversion
+	planDiscountedRawPrice: number; // discounted on site plan upgrade
+}
+
+interface Props {
+	planSlug: PlanSlug;
+	monthly?: boolean; // defaults to true
+}
+
+const usePlanPrices = ( { planSlug, monthly = true }: Props ): PlanPrices => {
```

Comment:
Let's create a follow-up issue to update raw/discount-price computations to use this hook, once we settle on merging this. 

We had put together something similar in the previous plans-comparison [here](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/plans-comparison/use-plan-prices.tsx) (albeit a bit outdated now), which for whatever reason we migrated away from or chose not to reuse (perhaps due to basing the 2023 grid work on the older plans framework).